storage/mysql/provider.go

```diff
@@ -81,7 +87,14 @@ func getMySQLDatabaseLocked() (*sql.DB, error) {
 	if mysqlDB != nil || mysqlErr != nil {
 		return mysqlDB, mysqlErr
 	}
-	db, err := OpenDB(*mySQLURI)
+	if err := registerTLSConfig(); err != nil {
+		return nil, err
+	}
```

Comment:
Could this be called inside the `if *mySQLTLSCA {` check on line 93?

That would make it clear that this is related to the MySQL TLS functionality.